### PR TITLE
Add EditModule to DG. Fix all diagrams to have valid activation bar

### LINKED ISSUE #310: Update Module Manager of DG

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

File: docs/diagrams/EditModuleSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+'end
+
+'edit tasks
+loop task in tasksToEdit
+    EditModuleCommand -> Model : editTask(task)
+    activate Model
+    Model -> TrackIter : editTask(task)
+deactivate Model
+end
+'end
```
Comment (simonteozw) [inline docs/diagrams/EditModuleSequenceDiagram.puml]: The way this is displayed in the diagram looks a bit odd. The Arrow doesnt seem to point correctly. Also, the return arrow for tasksToEdit and lessonstoEdit is not a dotted arrow

**Review by simonteozw — COMMENTED** [review]
Looks good! Just 1 comment

File: docs/diagrams/EditModuleSequenceDiagram.puml (hunk 1 above)
Comment (UncleGrandpa925) [reply docs/diagrams/EditModuleSequenceDiagram.puml]: Ohhh yeahhh. Okay lemme see how can I fix it

Comment (UncleGrandpa925) [reply docs/diagrams/EditModuleSequenceDiagram.puml]: Fixed! Can you review it again? Thanks!

**Review by simonteozw — APPROVED** [review]
LGTM

**Pull request merged**